aten/src/ATen/core/ivalue.cpp

```diff
@@ -118,4 +118,24 @@ void ivalue::Object::resizeObject(size_t slot) {
   slots_.resize(type()->numAttributes());
 }
 
+static bool CompareIValue(const std::pair<IValue, IValue>& aWrap,
+                          const std::pair<IValue, IValue>& bWrap) {
+  const auto a = aWrap.first;
+  const auto b = bWrap.first;
+  if (a.isString() && b.isString()) {
+    return a.toStringRef().compare(b.toStringRef()) < 0;
+  } else if (a.isInt() && b.isInt()) {
+    return a.toInt() < b.toInt();
+  } else if (a.isDouble() && b.isDouble()) {
+    return a.toDouble() < b.toDouble();
+  }
+  AT_ERROR("Illegal dict key");
+}
+
+const ivalue::GenericDict::IterationOrder ivalue::GenericDict::iterationOrder() const {
+    IterationOrder ordered(elements().begin(), elements().end());
+    std::sort(ordered.begin(), ordered.end(), CompareIValue);
+    return ordered;
```

Comment:
Starting from Python 3.7 there is a guarantee that all dicts are really ordered dicts, so we should probably be careful in the order we have selected here.